core/sdk-core/src/main/java/software/amazon/awssdk/core/SelectedAuthScheme.java

```diff
@@ -29,19 +29,32 @@
  */
 @SdkProtectedApi
 public final class SelectedAuthScheme<T extends Identity> {
```

Comment:
Should this be 2 classes instead, one that support signing and one that does not? It would avoid the big `if` in the constructor